Tasks/VsTest/vstest.ts

```diff
@@ -245,13 +247,25 @@ function publishCodeChanges(): Q.Promise<string> {
 
 function executeVstest(testResultsDirectory: string, parallelRunSettingsFile: string, vsVersion: number, argsArray: string[]): Q.Promise<number> {
     var defer = Q.defer<number>();
-    var vsCommon = tl.getVariable("VS" + vsVersion + "0COMNTools");
-    if (!vsCommon) {
-        tl.error(tl.loc('VstestNotFound', vsVersion));
-        defer.resolve(1);
-        return defer.promise;
+    if (vstestLocationMethod == "version") {
+        var vsCommon = tl.getVariable("VS" + vsVersion + "0COMNTools");
+        if (!vsCommon) {
+            tl.error(tl.loc('VstestNotFound', vsVersion));
+            defer.resolve(1);
+            return defer.promise;
+        }
+        vstestLocation = path.join(vsCommon, "..\\IDE\\CommonExtensions\\Microsoft\\TestWindow\\vstest.console.exe");
+    } else if (vstestLocationMethod == "location") {
+        try {
+            fs.accessSync(vstestLocation);
```

Comment:
Will fs.accessSync throw error if File Doesn't exist? if yes then file not there and access denied will lead to same exception